components/serverless/api/v1beta1/function_types.go

```diff
@@ -0,0 +1,206 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+type Runtime string
+
+const (
+	Python39 = "python39"
+	NodeJs14 = "nodejs14"
+)
+
+type Source struct {
+	// +optional
+	ArtifactSource *ArtifactSource `json:"artifactSource"`
+	//TODO: we should check if it is possible
+	// +optional
+	CRDSource *CRDSource `json:"crdSource"`
+	// +optional
+	GitRepositorySource *GitRepositorySource `json:"gitRepositorySource"`
+	// +optional
+	InlineSource *InlineSource `json:"inlineSource"`
+}
+
+type InlineSource struct {
+	Source       string `json:"source"`
+	Dependencies string `json:"dependencies"`
+}
+
+type GitRepositorySource struct {
+	Name      string `json:"name"`
+	Namespace string `json:"namespace"`
+}
+
+type CRDSource struct {
+	Api        string `json:"api"`
+	Version    string `json:"version"`
+	Kind       string `json:"kind"`
+	Name       string `json:"name"`
+	Namespaces string `json:"namespaces"`
```

Comment:
Do you thing it makes sense to use `ObjectMeta` and `TypeMeta` for these fields instead?

Reply:
Can those objects be used to reference particular crd?
I looked at them and they carry many information which I think we don't need.

Reply:
Could you give me a example of what those 2 objects expose and can be helpful?

Reply:
Nothing comes to my mind at this point.